pyathena/sqlalchemy/base.py

```diff
@@ -435,6 +435,29 @@ def limit_clause(self, select: "GenerativeSelect", **kw):
             text.append(" LIMIT " + self.process(select._limit_clause, **kw))
         return "\n".join(text)
 
+    def get_from_hint_text(self, table, text):
+        """Gets the text of a FROM clause hint"""
+        return text
+
+    def format_from_hint_text(self, sqltext, table, hint, iscrud):
+        """ Parses the text of a FROM clause hint, returning the text formatted properly"""
+        hint_upper = hint.upper()
+
+        if not any([
+            hint_upper.startswith("FOR TIMESTAMP AS OF"),
+            hint_upper.startswith("FOR SYSTEM_TIME AS OF"),
+            hint_upper.startswith("FOR VERSION AS OF"),
+            hint_upper.startswith("FOR SYSTEM_VERSION AS OF"),
+        ]):
+            raise exc.CompileError("Unrecognized hint: %r" % hint)
+
+        if 'AS' in sqltext.upper():
+            _, alias = sqltext.upper().split(' AS ', 1)
+            return f"{table.original.fullname} {hint} AS {alias.lower()}"
+
+        return f"{sqltext} {hint}"
```

Comment:
https://github.com/sqlalchemy/sqlalchemy/blob/414025a631904f4f3c767a055a8265b9cbb9407a/lib/sqlalchemy/sql/compiler.py#L4599
```suggestion
        if hinttext:
            if 'AS' in sqltext.upper() and any([
                hint_upper.startswith("FOR TIMESTAMP AS OF"),
                hint_upper.startswith("FOR SYSTEM_TIME AS OF"),
                hint_upper.startswith("FOR VERSION AS OF"),
                hint_upper.startswith("FOR SYSTEM_VERSION AS OF"),
            ]):
                _, alias = sqltext.upper().split(' AS ', 1)
                sqltext = f"{table.original.fullname} {hint} AS {alias}"
            else:
                sqltext = f"{sqltext} {hint}"
        return sqltext
```

Reply:
It is better to process the addition only when there is a hint.
alias should remain original and should not be converted to lower case.

Reply:
Is there any other way to determine if alias is specified? I don't think the following method is a good one.
```
_, alias = sqltext.upper().split(' AS ', 1)
```

Reply:
I thought we'd only get here if we were processing hints due to how the SQLAlchemy SQLCompiler used it. https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/sql/compiler.py#L4303-L4306

Reply:
As for the alias, we could check for an alias name on the table object which will be anon_1. Not sure we have a better way to get the real alias name, though, as it's part of an ambiguous mapper I don't think I have access to in this function. https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/sql/compiler.py#L5250-L5262

Reply:
https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/sql/compiler.py#L4391-L4392
AS will be uppercase unless the specification for this method changes. It is better since it does not do case conversion and splits with ` AS ` uppercase.

Reply:
Thank you, adjusted.